### PR TITLE
Make hikari connection pool timeout configurable

### PR DESCRIPTION
**Pull request by dasormeter**
Make hikari connection pool timeout configurable (default remains at 250.millis as it was previously)

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala [hunk 1]
```diff
@@ -1143,7 +1143,7 @@ private[platform] object JdbcLedgerDao {
         serverRole,
         jdbcUrl,
         connectionPoolSize,
-        250.millis,
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala:1146]: 250 millis is the minimum setting, hikari does not allow lower, if we need to increase this in some deployment scenarios that means that the query load + max connection pool size + this timeout combination is insufficient, (and I guess: increasing the connection pool size is not desirable).
So how about making this also a configuration parameter instead, so we can play around with this parameters per deployment?

Comment (dasormeter) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala:1146]: alright in the next commit extracted this value out to something configurable -- it does lead to a lot of repetition of the configuration in multiple places (we may be able to do further defaulting). https://github.com/digital-asset/daml/pull/9750/commits/c193e875b62cfaffca5242a20aec28242b07e4e8

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ParticipantConfig.scala [hunk 2]
```diff
@@ -32,6 +35,7 @@ object ParticipantConfig {
     s"jdbc:h2:mem:$participantId;db_close_delay=-1;db_close_on_exit=false"
 
   val DefaultManagementServiceTimeout: Duration = Duration.ofMinutes(2)
+  val DefaultApiServerDatabaseConnectionTimeout: Duration = Duration.ofMinutes(2)
```
Comment (nmarton-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ParticipantConfig.scala:38]: 250 millis perhaps?

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala (hunk 1 above)
Comment (nmarton-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala:1146]: wow, that exploded! sorry for that and thanks for doing it :raised_hands:

**Review by nmarton-da — APPROVED** [review]
Looks excellent to me :bow: 
Made one comment on a two minutes dbtimeout (probably a cpypastetypo)

**Pull request merged**